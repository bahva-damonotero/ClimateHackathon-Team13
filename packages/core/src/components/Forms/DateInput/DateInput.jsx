import React from 'react';
import PropTypes from 'prop-types';

import { Fieldset, FormError, Label, Legend } from '..';
import { Hint } from '../..';

export default function DateInput({
  children,
  disabled,
  error,
  hint,
  id,
  onChange,
  readOnly,
  value
}) {
  const hintId = `${id}--hint`;
  const monthId = `${id}--month`;
  const dayId = `${id}--day`;
  const yearId = `${id}--year`;

  const usedValue = {
    month: (value && value.month) || '',
    day: (value && value.day) || '',
    year: (value && value.year) || ''
  };

  return (
    <Fieldset>
      <Legend error={error}>{children}</Legend>

      {hint && <Hint id={hintId}>{hint}</Hint>}

      <FormError error={error} />

      <div className="usa-memorable-date">
        <div className="usa-form-group usa-form-group--month">
          <Label error={error} target={monthId}>
            Month
          </Label>

          <input
            aria-describedby={hint ? hintId : undefined}
            className="usa-input usa-input--inline"
            disabled={disabled}
            id={monthId}
            inputMode="numeric"
            maxLength="2"
            name={monthId}
            onChange={
              onChange
                ? (event) => onChange({ month: event.target.value })
                : undefined
            }
            pattern="[0-9]*"
            readOnly={readOnly}
            type="text"
            value={usedValue.month}
          />
        </div>

        <div className="usa-form-group usa-form-group--day">
          <Label error={error} target={dayId}>
            Day
          </Label>

          <input
            aria-describedby={hintId}
            className="usa-input usa-input--inline"
            disabled={disabled}
            id={dayId}
            inputMode="numeric"
            maxLength="2"
            name={dayId}
            onChange={
              onChange
                ? (event) => onChange({ day: event.target.value })
                : undefined
            }
            pattern="[0-9]*"
            readOnly={readOnly}
            type="text"
            value={usedValue.day}
          />
        </div>

        <div className="usa-form-group usa-form-group--year">
          <Label error={error} target={yearId}>
            Year
          </Label>

          <input
            aria-describedby={hintId}
            className="usa-input usa-input--inline"
            disabled={disabled}
            id={yearId}
            inputMode="numeric"
            minLength="4"
            maxLength="4"
            name={yearId}
            onChange={
              onChange
                ? (event) => onChange({ year: event.target.value })
                : undefined
            }
            pattern="[0-9]*"
            readOnly={readOnly}
            type="text"
            value={usedValue.year}
          />
        </div>
      </div>
    </Fieldset>
  );
}

DateInput.propTypes = {
  /** Label for input */
  children: PropTypes.node.isRequired,
  /** Should the inputs be marked as disabled? */
  disabled: PropTypes.bool,
  /** Error message to display */
  error: PropTypes.string,
  /** Hint to be shown for the date input */
  hint: PropTypes.node,
  /** ID of the input */
  id: PropTypes.string.isRequired,
  /**
   * Function called when one of the inputs
   * is modified
   */
  onChange: PropTypes.func,
  /** Should the inputs be marked as read-only? */
  readOnly: PropTypes.bool,
  /** Used to set the values of the inputs */
  value: PropTypes.shape({
    day: PropTypes.number,
    month: PropTypes.number,
    year: PropTypes.number
  })
};

DateInput.defaultProps = {
  disabled: false,
  readOnly: false
};
