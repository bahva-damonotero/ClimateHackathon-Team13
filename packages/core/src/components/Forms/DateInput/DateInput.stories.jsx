import React from 'react';

import Form from '../Form';
import DateInput from '.';

export default {
  title: 'Forms/DateInput',
  component: DateInput
};

const Template = (args) => (
  <Form>
    <DateInput {...args} />
  </Form>
);

const defaultArgs = {
  children: 'Label',
  id: 'dateinput-example'
};

export const def = Template.bind({});
def.storyName = 'Default';
def.args = {
  ...defaultArgs
};

export const hint = Template.bind({});
hint.args = {
  ...defaultArgs,
  hint: 'For example: 04 28 1986'
};

export const value = Template.bind({});
value.args = {
  ...defaultArgs,
  value: { day: 1, month: 1, year: 1970 },
  readOnly: true
};

export const error = Template.bind({});
error.args = {
  ...defaultArgs,
  error: 'Error with form field(s)'
};
