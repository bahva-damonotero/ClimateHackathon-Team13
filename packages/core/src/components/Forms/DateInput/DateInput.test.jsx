import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import DateInput from '.';

describe('DateInput', () => {
  const id = 'date-input-example';

  it('should call `onChange` with date data that was changed', () => {
    const onChange = jest.fn();

    const { container } = render(
      <DateInput id={id} onChange={onChange}>
        Label
      </DateInput>
    );

    fireEvent.change(container.querySelector(`#${id}--month`), {
      target: { value: 10 }
    });
    expect(onChange.mock.calls[0][0]).toEqual({ month: '10' });

    fireEvent.change(container.querySelector(`#${id}--day`), {
      target: { value: 25 }
    });
    expect(onChange.mock.calls[1][0]).toEqual({ day: '25' });

    fireEvent.change(container.querySelector(`#${id}--year`), {
      target: { value: 1995 }
    });
    expect(onChange.mock.calls[2][0]).toEqual({ year: '1995' });
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <DateInput
          hint="For example: 04 28 1986"
          id={id}
          readOnly
          value={{ month: 10, day: 25, year: 1995 }}
        >
          Label
        </DateInput>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
