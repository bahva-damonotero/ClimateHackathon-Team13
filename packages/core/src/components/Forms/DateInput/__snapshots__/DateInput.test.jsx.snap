// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput should match snapshot 1`] = `
<fieldset
  className="usa-fieldset"
>
  <legend
    className="usa-legend"
  >
    Label
  </legend>
  <span
    className="usa-hint"
    id="date-input-example--hint"
  >
    For example: 04 28 1986
  </span>
  <div
    className="usa-memorable-date"
  >
    <div
      className="usa-form-group usa-form-group--month"
    >
      <label
        className="usa-label"
        htmlFor="date-input-example--month"
      >
        Month
      </label>
      <input
        aria-describedby="date-input-example--hint"
        className="usa-input usa-input--inline"
        disabled={false}
        id="date-input-example--month"
        inputMode="numeric"
        maxLength="2"
        name="date-input-example--month"
        pattern="[0-9]*"
        readOnly={true}
        type="text"
        value={10}
      />
    </div>
    <div
      className="usa-form-group usa-form-group--day"
    >
      <label
        className="usa-label"
        htmlFor="date-input-example--day"
      >
        Day
      </label>
      <input
        aria-describedby="date-input-example--hint"
        className="usa-input usa-input--inline"
        disabled={false}
        id="date-input-example--day"
        inputMode="numeric"
        maxLength="2"
        name="date-input-example--day"
        pattern="[0-9]*"
        readOnly={true}
        type="text"
        value={25}
      />
    </div>
    <div
      className="usa-form-group usa-form-group--year"
    >
      <label
        className="usa-label"
        htmlFor="date-input-example--year"
      >
        Year
      </label>
      <input
        aria-describedby="date-input-example--hint"
        className="usa-input usa-input--inline"
        disabled={false}
        id="date-input-example--year"
        inputMode="numeric"
        maxLength="4"
        minLength="4"
        name="date-input-example--year"
        pattern="[0-9]*"
        readOnly={true}
        type="text"
        value={1995}
      />
    </div>
  </div>
</fieldset>
`;
